### PR TITLE
Equinix Provider - Fix Provisioning Error Checks

### PR DESCRIPTION
**Pull request by cprivitere**
This PR fixes some issues with the Equinix Provider. 

* We had a typo where the environ.go Instance check that was causing a nil pointer reference and thus it couldn't properly check instance statuses.
* waitDeviceActive was manually passing a nil value instead of an error all the time. We now return the device and error all the time so that the calling functions can properly handle the errors. This permits failed provisioning to be detected and retried.
* Add a test to TestWaitDeviceActive to make sure we get an error message back when provisioning fails.
* Updated instance.go's Status() to use our actual status names when checking device state.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
~- [ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
~- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

There is not a simple way to QA this one due to needing to simulate a provisioning failure. We tested this on our end using some debugging tools to manually fail an instance during provisioning and verified that juju now retries provisioning instead of waiting forever.

## Documentation changes

*How it affects user workflow (CLI or API). Delete section if not applicable.*

## Bug reference

*Link to Launchpad bug. Delete section if not applicable.*

### CODE REVIEW

File: provider/equinix/environ.go [hunk 1]
```diff
@@ -808,11 +808,7 @@ func waitDeviceActive(ctx context.ProviderCallContext, c *packngo.Client, id str
 		Clock:    clock.WallClock,
 	})
 
-	if d != nil {
-		return d, nil
-	}
-
-	return nil, err
+	return d, err
```
Comment (manadart) [inline provider/equinix/environ.go:811]: Please use `errors.Trace`, `errors.Annotate` or `errors.Annotatef`. These will ensure errors include stack traces, and are fine to call on `nil` errors, which just return nil.

File: provider/equinix/instance.go [hunk 2]
```diff
@@ -52,12 +52,12 @@ func (device *equinixDevice) Status(ctx context.ProviderCallContext) instance.St
...
 	switch device.State {
-	case "provisioning":
+	case "provisioning", "queued":
 		jujuStatus = status.Pending
 	case "active":
 		jujuStatus = status.Running
-	case "shutting-down", "terminated", "stopping", "stopped":
-		jujuStatus = status.Empty
+	case "deleted", "deprovisioning", "failed", "inactive":
+		jujuStatus = status.ProvisioningError
```
Comment (SimonRichardson) [inline provider/equinix/instance.go:60]: I think these should be typed strings, making it easier to reason about.

Comment (cprivitere) [reply provider/equinix/instance.go:60]: Can you elaborate what you're looking for here? Maybe an example from another provider. I saw the Azure one but I actually find it much harder to read/reason than what we have here.

Comment (SimonRichardson) [reply provider/equinix/instance.go:60]: LXD is a good example here: https://github.com/juju/juju/blob/develop/provider/lxd/instance.go#L38-L56

Comment (cprivitere) [reply provider/equinix/instance.go:60]: Ok, so unlike LXD, the packngo library doesn't have an enum for these statuses. So I'll need to add one to this file.

**Review by manadart — APPROVED** [review]

**Pull request merged**